packages/app/client/src/ui/pages/Register/index.js

```diff
@@ -59,28 +78,77 @@ function Register () {
         password
       })
       changeStep(2)
-    } catch(error) {
-      showToast({ message: error.message, severity: 'danger' }) 
+    } catch (error) {
+      showToast({ message: error.message, severity: 'danger' })
     }
   }
 
   const showCreateAccount = step <= 1
 
   return (
     <MainContainer size="small">
-      {showCreateAccount ? 
+      {showCreateAccount ? (
         <>
           <SimpleFormWithHeader
             onSubmit={step === 0 ? verifyEmail : createAccount}
             {...formProps[step]}
-            subtitle={isMobileViewPort ? "" : formProps[step].subtitle}
+            subtitle={isMobileViewPort ? '' : formProps[step].subtitle}
           />
           {step === 0 && <CreateAccountFooter />}
-        </> :
-        <EmailSent email={verifiedEmail} />
-      }
+        </>
+      ) : (
+        <EmailSent email={verifiedEmail} action={sendEmailToSignUp} />
+      )}
     </MainContainer>
   )
 }
 
-export default Register
\ No newline at end of file
+const mapStateToProps = state => ({
+  auth: state.auth
+})
+
+const mapDispatchToProps = dispatch => ({
```

Comment:
hey man, we are using the Redux Hooks, I think they will save you from all this work with that two HOC. You could use the useDispatch to the actions and the useSelector to get things from the state. Actually, we have a Selector folder, where you will find what you need in terms of getting data.